src/language-usability.rst

```diff
@@ -16,6 +16,193 @@ Usability of Type Systems
 API Usability
 =============
 
+..
+  Outline
+  * APIs are ubiquitous
+  * Examples
+  * High level goals of API
+  * Difficulty in achieving goals of API
+  * Distinction with language usability
+..
+
+Application Programming Interfaces (APIs) are interfaces that mediate interactions between software components.
+Examples include software libraries and frameworks, OS system calls, and web protocols.
+Nearly every program written will use an API :cite:`myers2016improving`.
+
+The goal of the API designer is to create an API that a broad range of
+programmers can use in their applications.
+This goal mirrors the goals of programming language designers --
+an API designer must a create a service that is "usable" to
+a broad and vaguely scoped audience.
+Like languages, APIs must be simple enough to be used
+by users who are unaware of the inner workings of the API.
+
+In many cases, both language and API design can be approached through a "bottom-up" manner -- designed with smaller components
```

Comment:
APIs differ from PLs in that PLs are axiomatic: a language definition defines a universe of programs. APIs operate within a programming language by making use of its primitive operations as well as other APIs.